database/dataGeneration.js

```diff
@@ -0,0 +1,133 @@
+const faker = require('faker/locale/en_US');
+const Jabber = require('jabber');
+const moment = require('moment');
+const fs = require('fs');
+
+const buzzWords = ['great food', 'eat', 'service', 'drinks', 'loved', 'never again', 'dessert', 'main menu', 'appetizers', 'enjoyed', 'hated', 'server', 'noise', 'waitress', 'waiter', 'slow', 'fast', 'go again', 'clean', 'dirty', 'kids', 'our group', 'slow service', 'cold food', 'awesome drinks', 'fun time', 'romantic atmosphere', 'dinner', 'lunch', 'breakfast', 'hot plate', 'smelled wonderful', 'no alcohol', 'bartender', 'wine list', 'outdoor seating', 'large room', 'no tips', 'valet parking', 'fine dining', 'dishes', 'shrimp', 'steak', 'chicken', 'beef', 'pasta', 'cake', 'pastries', 'creme brulee', 'simmering', 'coffee', 'fresh food', 'candlelit', 'anniversary', 'birthday', 'party', 'with friends', 'evening', 'stuffed', 'would go again', 'recommend this place', 'restrooms clean', 'host greeting', 'lovely place', 'best view', 'indoor seating', 'waited an hour', 'risotto', 'baked salmon', 'fried rice', 'shrimp scampi', 'sushi', 'wait time'];
+
+const tags = ['Good for groups', 'Desserts', 'Appetizers', 'Drinks', 'Kid friendly'];
+
+const circleColors = ['#6c8ae4', '#d86441', '#bb6acd', '#df4e96'];
+
+const jabber = new Jabber(buzzWords, 2);
+
+const allWritesFinished = [false, false];
+
+
+
+async function createReviewersAndReviews(NumberOfPeople) {
+  let reviewerCount = NumberOfPeople || 100;
+  const writeReviews = fs.createWriteStream('Reviews.csv');
+  const writeReviewers = fs.createWriteStream('Reviewers.csv');
+  let ok = true;
+  let reviewIndex = 0;
+
+
+  writeReviewers.write('id, color, vip, reviewCount, firstName, lastName \n');
+  writeReviews.write(`id, restaurantId, reviewerId overall, food, service, ambience, dineDate, noise, recommend, comments, filterTag \n`)
+
+  writeReviewerToCSV();
+  async function writeReviewerToCSV() {
+    do {
+      let reviewer = {};
+      reviewer.id = reviewerCount;
+      reviewer.reviewCount = faker.random.number({min: 1, max: 15})
+      reviewer.color = faker.random.arrayElement(circleColors);
+      reviewer.vip = Math.random() < 0.3;
+      reviewer.firstName = (Math.random() < 0.7) ? faker.name.firstName() : '';
+      if (reviewer.firstName.length) {
+        reviewer.lastName = (Math.random() < 0.7) ? faker.name.lastName() : '';
+      } else {
+        reviewer.lastName = '';
+      }
+      reviewerCount -= 1; // maybe move to below the review section?
+
+      ///////////////////////////////////////////
+
+      reviewIndex += await writeReview(reviewer.id, reviewer.reviewCount, writeReviews, reviewIndex);
+
```

Comment:
This is an interesting use of await, I can't wait to see what happens with the massive amount of data you're generating.

Reply:
I can tell you now, it won't end well.